src/user_input_compiler.rs

```diff
@@ -110,6 +112,10 @@ fn match_command_word(s :&mut Chars) -> Input {
     return Grammar;
   } else 
   
+  if starts_with(s, "location") {
+    return Location;
+  } else 
```

Comment:
Instead of this if-elseif construct, it would probably make sense to tokenize the input and to match on the first token.

But that's something for another PR :)